### PR TITLE
V2-telemetry: Simplify sensitive metadata allowlist to accept feature only

### PR DESCRIPTION
**Pull request by akalia25**
This PR adds another option to the sensitive metadata allowlist by simplifying the input requirements. to accept `feature`  and the allowlisted `privateMetadata` keys for that feature. 

This change is particularly beneficial when a `feature` has multiple associated `action`(s), and the `privateMetadata` key needs to be allowed for all events related to that feature.

Building upon this initial PR:
- https://github.com/sourcegraph/sourcegraph/pull/62830/files

## Test plan

CI and unit tests

### CODE REVIEW

**Comment by bobheadxi** [comment]
@akalia25 ah, I misunderstood - I was hoping we could make action _optional_, not _always omitted_. Could we do that instead here?

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go [hunk 1]
```diff
@@ -37,39 +37,123 @@ func AllowedEventTypes() EventTypes {
...
 			},
 		},
-		// The 'languageId' key is included for feature:'cody.completions' action:suggested/accepted events to provide
+		// The 'languageId' key is included for feature:'cody.completions' events to provide
 		// customers with valuable language-specific insights from the analytics we offer.
 		// This information helps them better understand code completion usage patterns.
 		EventType{
 			Feature: "cody.completion",
-			Action:  "suggested",
+			Action:  "%",
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:45]: Conventional wildcard format is `*`:

```suggestion
			Action:  "*",
```

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go [hunk 2]
```diff
@@ -37,39 +37,123 @@ func AllowedEventTypes() EventTypes {
...
+	// index of '{feature}:{action}:{allowedfields}' for checking
 	index map[string][]string
 }
 
 func eventTypes(types ...EventType) EventTypes {
 	index := make(map[string][]string, len(types))
 	for _, t := range types {
-		index[fmt.Sprintf("%s.%s", t.Feature, t.Action)] = t.AllowedPrivateMetadataKeys
+		key := fmt.Sprintf("%s.%s", t.Feature, t.Action)
+		if t.Action == "%" {
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:153]:
```suggestion
		if t.Action == "*" {
```

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go [hunk 3]
```diff
@@ -90,9 +174,22 @@ func (e EventTypes) Redact(event *telemetrygatewayv1.Event) redactMode {
 // IsAllowed indicates an event is on the sensitive telemetry allowlist, and the fields that
 // are allowed.
 func (e EventTypes) IsAllowed(event *telemetrygatewayv1.Event) ([]string, bool) {
-	key := fmt.Sprintf("%s.%s", event.GetFeature(), event.GetAction())
-	allowedKeys, allowed := e.index[key]
-	return allowedKeys, allowed
+	var key string
+	if event.GetAction() == "%" {
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:178]:
```suggestion
	if event.GetAction() == "*" {
```

Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:178]: The incoming event's action will never be wildcard

Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:178]: This code should instead check, "is this feature alone, on the allowlist?"

Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:178]: We'll want to account for overlaps as well, preferring the more granular one - for example, for event `foo.bar`, if allowlist entry `foo.*` and `foo.bar` exists, then the `foo.bar` one takes precedence

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go [hunk 4]
```diff
@@ -37,39 +37,123 @@ func AllowedEventTypes() EventTypes {
...
-	// index of '{feature}.{action}:{allowedfields}' for checking
+	// index of '{feature}:{action}:{allowedfields}' for checking
 	index map[string][]string
 }
 
 func eventTypes(types ...EventType) EventTypes {
 	index := make(map[string][]string, len(types))
 	for _, t := range types {
-		index[fmt.Sprintf("%s.%s", t.Feature, t.Action)] = t.AllowedPrivateMetadataKeys
+		key := fmt.Sprintf("%s.%s", t.Feature, t.Action)
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:152]: In hindsight this index should use another character that isn't a legal feature/action character, to avoid situations where:

```
feature: a.b, action: c
feature: a, action: b.c
```

resolves to the same entry. the `/` character might be a better one to use here:

```suggestion
		key := fmt.Sprintf("%s/%s", t.Feature, t.Action)
```

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go [hunk 5]
```diff
@@ -90,9 +174,22 @@ func (e EventTypes) Redact(event *telemetrygatewayv1.Event) redactMode {
...
-	return allowedKeys, allowed
+	var key string
+	if event.GetAction() == "%" {
+		key = event.GetFeature()
+		allowedKeys, allowed := e.index[key]
+		if allowed {
+			return allowedKeys, allowed
+		}
+	} else {
+		key = fmt.Sprintf("%s.%s", event.GetFeature(), event.GetAction())
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:185]: match https://github.com/sourcegraph/sourcegraph/pull/63325/files#r1649431906

```suggestion
		key = fmt.Sprintf("%s/%s", event.GetFeature(), event.GetAction())
```

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go (hunk 3 above)
Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:178]: Pseudocode:

```
if e.index[event.Feature / event.Action] {
   // ... use full feature/action allowlist entry
} else if e.index[event.Feature] {
   // ... use feature-wide allowlist entry
}
```

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go (hunk 1 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go]: ahh I was thinking with my SQL brain. Updated, thank you!

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go (hunk 2 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go]: updated, thank you!

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go (hunk 3 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go]: good catch! I updated, and with the order of operations, this:
```
We'll want to account for overlaps as well, preferring the more granular one - for example, for event foo.bar, if allowlist entry foo.* and foo.bar exists, then the foo.bar one takes precedence
```
Should be accounted for. It'll first always check for the feature/action allowed keys first.

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go (hunk 4 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go]: updated to use `/` :)

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go (hunk 5 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go]: updated!

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 6]
```diff
@@ -126,15 +133,13 @@ func TestParseAdditionalAllowedEventTypes(t *testing.T) {
 func TestEventTypesRedact(t *testing.T) {
 	allowedTypes := eventTypes(EventType{
 		Feature:                    "example",
-		Action:                     "exampleAction",
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:129]: Removing this should cause a failure right? Empty value is unsupported, only explicit wildcard is supported

Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:129]: Either way, these tests should remain unchanged to ensure the existing behaviour works. What should happen instead is you add a few _new_ test cases that demonstrate the wildcard behaviour

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 7]
```diff
@@ -41,10 +41,21 @@ func TestIsAllowed(t *testing.T) {
...
 		},
 		{
-			name: "disallowed event with additional allowed event type",
+			name: "disallowed event with additional allowed event type (feature only)",
 			event: &v1.Event{
 				Feature: "cody.completion",
-				Action:  "accepted",
+				Action:  "*",
+			},
+			expectAllowed: true,
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:49]: This might be a carry-over from what was here before, but the test case says "disallowed event" but the result is `expectAllowed: true`, it sounds like the test name is incorrect?

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 6 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:129]: [this](https://github.com/sourcegraph/sourcegraph/pull/63325/files#diff-b61c40a8ec14aafd5f7584a33f4fb07acf6067d60b16c0219fc674fd43747a7aL44-R58) is the test I added for the wildcard behavior, should I add more?

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 8]
```diff
@@ -41,10 +41,21 @@ func TestIsAllowed(t *testing.T) {
...
+				"languageId",
+			},
+		},
+		{
+			name: "disallowed event with additional allowed event type",
+			event: &v1.Event{
+				Feature: "cody.hoverCommands",
+				Action:  "visible",
 			},
 			expectAllowed: true,
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:60]: same here, the test case says "disallowed event" but the result is `expectAllowed: true`

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 6 above)
Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:129]: That one tests "is allowed", this test is `TestEventTypesRedact` which tests redaction. If redaction behaviour is unchanged, then the tests should remain unchanged as well

(i.e. just remove all your changes to this test :) )

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 7 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:49]: yeah, I think the test name is confusing. I tried keeping the same convention, and had to interpret it like this event has `privateMetadata` ("disallowed event") and there is an allowlist of keys ("with additional allowed event type"). I'll update to make more clear?

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 6 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:129]: ahhh whoops 🙈 sorry, got the tests mixed up. No change to the redaction logic. I'll remove this. Thank you!

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 7 above)
Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:49]: I think the test case should say, "event with private metadata, that has fields it is allwoed to export"

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 6 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go]: Removed the changes I had for `TestEventTypesRedact` and added a new test `allowlisted, with wildcard(*) action match` and the expected redaction it should do. Let me know if it's unnecessary and I can remove that . Thanks again!

Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go]: I think it's a good test to have, thanks! Can you add an assertion on the resulting data?

Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go]: the function directly mutates the value you give it

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 9]
```diff
@@ -82,30 +93,26 @@ func TestParseAdditionalAllowedEventTypes(t *testing.T) {
...
 					},
 				},
 				{
-					Feature: "baz.bar",
-					Action:  "bar.baz",
-					AllowedPrivateMetadataKeys: []string{
-						"field",
-					},
+					Feature:                    "baz.bar",
+					AllowedPrivateMetadataKeys: []string{"field"},
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:115]: Shouldn't the expected result here have `Action: "*"`?

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go [hunk 10]
```diff
@@ -37,39 +37,123 @@ func AllowedEventTypes() EventTypes {
...
+				"model",
+			},
+		},
 	)...)
 }
 
 type EventTypes struct {
 	types []EventType
-	// index of '{feature}.{action}:{allowedfields}' for checking
+	// index of '{feature}:{action}:{allowedfields}' for checking
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:145]:
```suggestion
	// index of '{feature}/{action}:{allowedfields}' for checking
```

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go [hunk 11]
```diff
@@ -149,11 +243,19 @@ func parseAdditionalAllowedEventTypes(config string) ([]EventType, error) {
...
+				Feature:                    parts[0],
+				AllowedPrivateMetadataKeys: parts[2:],
+			})
+		} else {
+			types = append(types, EventType{
+				Feature:                    parts[0],
+				Action:                     parts[1],
+				AllowedPrivateMetadataKeys: parts[2:],
+			})
+		}
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:258]: I don't think this is needed because we should have `Action: "*"` explicitly right?

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 7 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go]: updated with new naming :)

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 8 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go]: updated with new naming

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go (hunk 11 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go]: Correct! this is not needed, and this was also affecting [this](https://github.com/sourcegraph/sourcegraph/pull/63325#discussion_r1651531264), because we would not get back `action: *` on EventType. Updated this. Thank you!

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 12]
```diff
@@ -186,5 +194,32 @@ func TestEventTypesRedact(t *testing.T) {
...
+			})
+			mode := allowedTypes.Redact(&v1.Event{
+				Feature: "example",
+				Action:  "randomExampleAction",
+			})
+			ev := &v1.Event{
+				Parameters: &v1.EventParameters{
+					PrivateMetadata: &structpb.Struct{
+						Fields: map[string]*structpb.Value{
+							"foo": {
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:211]:
```suggestion
							"notAllowed": ...
							"foo": {
```

Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:211]: added this test and updated the field types in the test to be string value for extra measure :)

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go [hunk 13]
```diff
@@ -148,12 +242,13 @@ func parseAdditionalAllowedEventTypes(config string) ([]EventType, error) {
...
+				Feature:                    parts[0],
+				Action:                     parts[1],
+				AllowedPrivateMetadataKeys: parts[2:],
+			})
 		}
-		types = append(types, EventType{
-			Feature:                    parts[0],
-			Action:                     parts[1],
-			AllowedPrivateMetadataKeys: parts[2:],
-		})
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go:156]: Is this change needed? The `if` case above already returns early. Not indenting unnecessarily is good for readability IMO

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 14]
```diff
@@ -186,5 +194,38 @@ func TestEventTypesRedact(t *testing.T) {
...
+								},
+							},
+						},
+					},
+				}}
+
+			assert.Equal(t, redactMarketingAndUnallowedPrivateMetadataKeys, mode)
+			// assert the values are still there for privateMetadata
+			assert.Equal(t, "allowed", ev.Parameters.PrivateMetadata.Fields["foo"].GetStringValue())
+			assert.Equal(t, "redacted", ev.Parameters.PrivateMetadata.Fields["bar"].GetStringValue())
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:228]: Hm, is this expected? It should be redacted right, i.e. the field shouldn't exist because it's not on the allowlist?

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 15]
```diff
@@ -186,5 +194,38 @@ func TestEventTypesRedact(t *testing.T) {
...
+				Feature:                    "example",
+				Action:                     "*",
+				AllowedPrivateMetadataKeys: []string{"foo"},
+			})
+			mode := allowedTypes.Redact(&v1.Event{
+				Feature: "example",
+				Action:  "randomExampleAction",
+			})
+			ev := &v1.Event{
+				Parameters: &v1.EventParameters{
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:208]: I think you want to define `ev` _before_ `mode := allowedTypes.Redact` on L203, and then change that to `mode := allowedTypes.Redact(ev)`

Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:208]: Otherwise, the test assertions below aren't actually testing against the mutated `ev` that goes through the redaction

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 14 above)
Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:228]: See https://github.com/sourcegraph/sourcegraph/pull/63325/files#r1655218884

Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go]: hmm you're right! This should not have passed, after updating the mode with `mode := allowedTypes.Redact(ev)` this assertion is now failing `assert.Equal(t, redactMarketingAndUnallowedPrivateMetadataKeys, mode)` Looking into it 👀

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go (hunk 13 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist.go]: updated, thank you!

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 15 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go]: Test have been updated, should be ready for hopefully the final review 🙏 thanks for your patience and review on this!

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 16]
```diff
@@ -186,5 +194,39 @@ func TestEventTypesRedact(t *testing.T) {
...
+					},
+				}}
+			mode := allowedTypes.Redact(ev)
+
+			assert.Equal(t, redactMarketingAndUnallowedPrivateMetadataKeys, mode)
+
+			// assert that only the allowlisted privateMetadata key (foo) has a value
+			assert.Equal(t, "allowed", ev.Parameters.PrivateMetadata.Fields["foo"].GetStringValue())
+			assert.Nil(t, ev.Parameters.PrivateMetadata.Fields["bar"])
+
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:229]: (extra whitespace)

```suggestion
```

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 17]
```diff
@@ -41,10 +41,21 @@ func TestIsAllowed(t *testing.T) {
 			expectAllowed: false,
 		},
 		{
-			name: "disallowed event with additional allowed event type",
+			name: "event with private metadata, that has fields it is allowed to export (feature only)",
 			event: &v1.Event{
 				Feature: "cody.completion",
-				Action:  "accepted",
+				Action:  "*",
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:47]: This is the event that goes _into_ the "is allowed" check, I think there shouldn't be a wildcard here? We can note that this is defined as a wildcard in the "is allowed" check

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 18]
```diff
@@ -41,10 +41,22 @@ func TestIsAllowed(t *testing.T) {
 			expectAllowed: false,
 		},
 		{
-			name: "disallowed event with additional allowed event type",
+			name: "event with private metadata, that has fields it is allowed to export (feature only)",
+			// This feature's action is defined as a wildcard(*) in the "IsAllowed" check's list of known events
 			event: &v1.Event{
 				Feature: "cody.completion",
-				Action:  "accepted",
+				Action:  "Accepted",
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:48]: per naming convention:

```suggestion
				Action:  "accepted",
```

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go [hunk 19]
```diff
@@ -186,5 +195,38 @@ func TestEventTypesRedact(t *testing.T) {
...
+								Kind: &structpb.Value_StringValue{
+									StringValue: "redacted",
+								},
+							},
+						},
+					},
+				}}
+			mode := allowedTypes.Redact(ev)
+
+			assert.Equal(t, redactMarketingAndUnallowedPrivateMetadataKeys, mode)
```
Comment (bobheadxi) [inline internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:225]: Sorry for another spacing nit 😆 

```suggestion
			mode := allowedTypes.Redact(ev)
			assert.Equal(t, redactMarketingAndUnallowedPrivateMetadataKeys, mode)
```

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 18 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go:48]: ahh 🤦 thank you, sorry about that!

File: internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go (hunk 19 above)
Comment (akalia25) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go]: oh you're good! Do you have a linter you use, I'm surprised its not auto-corrected sometimes

Comment (bobheadxi) [reply internal/telemetry/sensitivemetadataallowlist/sensitivemetadataallowlist_test.go]: No, this is just eyeing it and see if it looks good to read :)

**Review by bobheadxi — APPROVED** [review]
Nice work!
